python/pyspark/errors/error_classes.py

```diff
@@ -15,1160 +15,15 @@
-    ]
-  },
-  "UNSUPPORTED_PARAM_TYPE_FOR_HIGHER_ORDER_FUNCTION": {
-    "message": [
-      "Function `<func_name>` should use only POSITIONAL or POSITIONAL OR KEYWORD arguments."
-    ]
-  },
-  "UNSUPPORTED_SIGNATURE": {
-    "message": [
-      "Unsupported signature: <signature>."
-    ]
-  },
-  "UNSUPPORTED_WITH_ARROW_OPTIMIZATION": {
-    "message": [
-      "<feature> is not supported with Arrow optimization enabled in Python UDFs. Disable 'spark.sql.execution.pythonUDF.arrow.enabled' to workaround."
-    ]
-  },
-  "VALUE_ALLOWED": {
-    "message": [
-      "Value for `<arg_name>` does not allow <disallowed_value>."
-    ]
-  },
-  "VALUE_NOT_ACCESSIBLE": {
-    "message": [
-      "Value `<value>` cannot be accessed inside tasks."
-    ]
-  },
-  "VALUE_NOT_ALLOWED": {
-    "message": [
-      "Value for `<arg_name>` has to be amongst the following values: <allowed_values>."
-    ]
-  },
-  "VALUE_NOT_ANY_OR_ALL": {
-    "message": [
-      "Value for `<arg_name>` must be 'any' or 'all', got '<arg_value>'."
-    ]
-  },
-  "VALUE_NOT_BETWEEN": {
-    "message": [
-      "Value for `<arg_name>` must be between <min> and <max>."
-    ]
-  },
-  "VALUE_NOT_NON_EMPTY_STR": {
-    "message": [
-      "Value for `<arg_name>` must be a non-empty string, got '<arg_value>'."
-    ]
-  },
-  "VALUE_NOT_PEARSON": {
-    "message": [
-      "Value for `<arg_name>` only supports the 'pearson', got '<arg_value>'."
-    ]
-  },
-  "VALUE_NOT_PLAIN_COLUMN_REFERENCE": {
-    "message": [
-      "Value `<val>` in `<field_name>` should be a plain column reference such as `df.col` or `col('column')`."
-    ]
-  },
-  "VALUE_NOT_POSITIVE": {
-    "message": [
-      "Value for `<arg_name>` must be positive, got '<arg_value>'."
-    ]
-  },
-  "VALUE_NOT_TRUE": {
-    "message": [
-      "Value for `<arg_name>` must be True, got '<arg_value>'."
-    ]
-  },
-  "VALUE_OUT_OF_BOUNDS": {
-    "message": [
-      "Value for `<arg_name>` must be between <lower_bound> and <upper_bound> (inclusive), got <actual>"
-    ]
-  },
-  "WRONG_NUM_ARGS_FOR_HIGHER_ORDER_FUNCTION": {
-    "message": [
-      "Function `<func_name>` should take between 1 and 3 arguments, but the provided function takes <num_args>."
-    ]
-  },
-  "WRONG_NUM_COLUMNS": {
-    "message": [
-      "Function `<func_name>` should take at least <num_cols> columns."
-    ]
-  },
-  "ZERO_INDEX": {
-    "message": [
-      "Index must be non-zero."
-    ]
-  }
-}
-'''
-
+# Note: Though we call them "error classes" here, the proper name is "error conditions",
+#   hence why the name of the JSON file different.
+#   For more information, please see: https://issues.apache.org/jira/browse/SPARK-46810
+#   This discrepancy will be resolved as part of: https://issues.apache.org/jira/browse/SPARK-47429
+# Note: When we drop support for Python 3.8, we should migrate from importlib.resources.read_text()
```

Comment:
We dropped Python 3.8 lately :-).

Reply:
Updated. I also reran test 5 from the PR description.